### DISCUSSION TITLE
IAM role access to s3 instead of access key

### DISCUSSION THREAD

**Discussion by jeffalgarne** [jhuckaby/Cronicle#608] [Q&A] [answered]
We are planning to use IAM role instead of access key to access our s3 bucket can cronicle config support this one?

**Comment by mikeTWC1984** [comment]
Did you mean managing user permission in IAM?

**Reply by jeffalgarne** [reply]
No, we wanted to use IAM user and policy to access the s3(For storage) bucket from cronicle.  We wanted to set one user and add the policy to it instead of using access key. Sample config below.

`{
	"engine": "S3",
	"AWS": {
		"region": "us-west-1",
		"credentials": {
			"accessKeyId": "YOUR_AMAZON_ACCESS_KEY", 
			"secretAccessKey": "YOUR_AMAZON_SECRET_KEY", 
		}
	},
	"S3": {
		"connectTimeout": 5000,
		"socketTimeout": 5000,
		"maxAttempts": 50,
		"keyPrefix": "",
		"fileExtensions": true,
		"params": {
			"Bucket": "MY_S3_BUCKET_ID"
		},
		"cache": {
			"enabled": true,
			"maxItems": 1000,
			"maxBytes": 10485760
		}
	}
}
`

**Comment by jhuckaby** [comment, marked as answer]
We use this method at my day job.  Your EC2 instances will automatically self-authenticate via the AWS SDK.  Just omit all the AWS credentials from your configuration file, and it will default to using EC2 (IAM) auth.

```json
"Storage": {
	"engine": "S3",
	"list_page_size": 50,
	"concurrency": 4,
	"log_event_types": { "get": 1, "put": 1, "head": 1, "delete": 1, "expire_set": 1 },
	
	"AWS": {
		"region": "us-west-1",
		"maxAttempts": 50,
		"connectTimeout": 5000,
		"socketTimeout": 5000
	},
	"S3": {
		"keyPrefix": "",
		"fileExtensions": true,
		"params": {
			"Bucket": "YOUR-S3-BUCKET-NAME-HERE"
		}
	}
},
```

**Reply by jeffalgarne** [reply]
@jhuckaby Thank you, will try this.
